### PR TITLE
add feature toggle behavior to common reducers; some clean up 

### PR DESCRIPTION
**Pull request by rianfowler**
# Description
- moves feature toggle behavior to common store
- move global selector for feature toggle state to site-wide folder
- refactor facility locator example to use global selector 
- add unit tests 

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/utilities/feature-toggles/feature-toggle-query-list.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "featureToggleQueryList": ["facilityLocatorShowCommunityCares"]
```
Comment (rianfowler) [inline src/platform/utilities/feature-toggles/feature-toggle-query-list.json]: After we add the feature toggle to the local environment Flipper admin, we'll start here and add the feature toggle string to this list. The Flipper API currently doesn't have a "get all" query so we have to tell it which feature toggle values we want returned.

File: src/applications/facility-locator/utils/selectors.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export const toggleValues = state => state.featureToggles || {};
-export const isProduction = state => toggleValues(state).production;
+import { toggleValues } from 'platform/site-wide/feature-toggles/selectors';
```
Comment (rianfowler) [inline src/applications/facility-locator/utils/selectors.js]: if you have a selector file you can import the toggle values selector here. You can see how this is being used below. 

We don't really use `reselect` heavily but this is a good use case for it. might refactor the toggleValues selector to use it so it gets memoized.

File: src/applications/facility-locator/containers/VAMap.jsx [hunk 3]
```diff
@@ -27,10 +27,8 @@ import ProviderMarker from '../components/markers/ProviderMarker';
 import { facilityTypes } from '../config';
 import { LocationType, FacilityType, BOUNDING_RADIUS } from '../constants';
 import { areGeocodeEqual /* areBoundsEqual */ } from '../utils/helpers';
-import {
-  isProduction,
-  facilityLocatorShowCommunityCares,
-} from '../utils/selectors';
+import { facilityLocatorShowCommunityCares } from '../utils/selectors';
+import { isProduction } from 'platform/site-wide/feature-toggles/selectors';
```
Comment (rianfowler) [inline src/applications/facility-locator/containers/VAMap.jsx]: not really necessary but we have an `isProduction` selector. The environment state is on the redux store now so we can use selectors for this if we want.

**Review by erikphansen — APPROVED** [review]

**Review by accbjt — APPROVED** [review]
LGTM

**Pull request merged**